src/core/server/environment/environment_service.test.ts

```diff
@@ -135,6 +135,32 @@ describe('UuidService', () => {
         expect(logger.get('process').warn).not.toHaveBeenCalled();
       });
     });
+
+    describe('unhandledRejection warnings', () => {
+      it('logs warn for an unhandeld promise rejected with an Error', async () => {
+        await service.preboot();
+
+        const err = new Error('something went wrong');
+        process.emit('unhandledRejection', err, new Promise((res, rej) => rej(err)));
+
+        expect(logger.get('process').warn).toHaveBeenCalledTimes(1);
+        expect(loggingSystemMock.collect(logger).warn[0][0]).toMatch(
+          /Detected an unhandled Promise rejection: Error: something went wrong\n.*at Object.<anonymous> \(/
```

Comment:
maybe we can remove the part after `at ` to avoid dependency on the serialization format?